### PR TITLE
[DOC-3514] Add warning about reusing an incompatible store directory

### PR DESCRIPTION
**Pull request by mdlinville**
[DOC-3514] Add warning about reusing an incompatible store directory

### CODE REVIEW

**Comment by srosenberg** [comment]
> [[DOC-3514](https://cockroachlabs.atlassian.net/browse/DOC-3514)] Add warning about reusing an incompatible store directory

There are existing checks in the code to bail out during an attempt to open a store with an incompatible version. I am curious what led to add an explicit warning. Is there a case where an attempt went undetected?

**Comment by mdlinville** [comment]
> > [[DOC-3514](https://cockroachlabs.atlassian.net/browse/DOC-3514)] Add warning about reusing an incompatible store directory
> 
> There are existing checks in the code to bail out during an attempt to open a store with an incompatible version. I am curious what led to add an explicit warning. Is there a case where an attempt went undetected?

Well, what did you have in mind when you filed the issue? :)

File: src/current/v22.2/start-a-local-cluster.md [hunk 1]
```diff
@@ -16,7 +16,11 @@ Once you've [installed CockroachDB](install-cockroachdb.html), it's simple to ru
...
+
+{{site.data.alerts.callout_danger}}
+A node's store directory contains cluster data, metadata, and logs. If the store is incompatible with the CockroachDB binary version, this can lead to panics or other problems when starting a node. If you previously started a cluster with a different major version of CockroachDB on a node, do not attempt to reuse the store location when starting a new cluster with a new CockroachDB version. Instead, either [upgrade the existing cluster]({% link {{ page.version.version }}/upgrade-cockroach-version.md %}) to the new version to update the store directory's contents, or move or delete the previous store directory before starting the `cockroach` process. The store directory is the `cockroach-data/` directory in the same directory as the `cockroach` command by default, or the location passed to the `--store` flag otherwise. For details about the store location, refer to [cockroach start]({% link {{ page.version.version }}/cockroach-start.md %}#store).
```
Comment (srosenberg) [inline src/current/v22.2/start-a-local-cluster.md:22]: I would rephrase the first two sentences as follows,

> If the previously initialized store is incompatible with either the new CockroachDB binary or the new cluster configuration (e.g., `--start-single-node` disables replication), this can lead to panics or other problems when starting a cluster. Instead, either move or delete the previous store directory before starting the `cockroach` process.

The stuff about `upgrade` is a bit more subtle. An incompatible binary would exit with an error message, i.e., not panic; otherwise, an auto upgrade kicks in. So, it's best to leave it underspecified. The warning is mainly about incompatible cluster configurations between runs.

**Review by srosenberg — APPROVED** [review]

File: src/current/v22.2/start-a-local-cluster.md (hunk 1 above)
Comment (mdlinville) [reply src/current/v22.2/start-a-local-cluster.md]: Thanks, I took your suggestion and reworked it a bit.

**Review by kathancox — APPROVED** [review]
LGTM, just some optional v small changes.
